File: java/src/main/java/com/leduy/quicktrace/examples/BasicExample.java
package com.leduy.quicktrace.examples;

import com.leduy.quicktrace.OutputStyle;
import com.leduy.quicktrace.Tracer;

/**
 * Basic usage với default style
 */
public class BasicExample {
    
    public static void main(String[] args) throws InterruptedException {
        // Basic usage với detailed style (tương tự Go version)
        Tracer tracer = Tracer.builder("Basic Example")
                .outputStyle(OutputStyle.DETAILED)
                .build();
        
        Thread.sleep(30);
        tracer.span("Initialize database");

        Thread.sleep(50);
        tracer.span("Load user data");

        Thread.sleep(20);
        tracer.span("Process data");

        Thread.sleep(10);
        tracer.span("Generate response");
        
        tracer.end();
    }
}

File: csharp/Examples/BasicExample.cs
using QuickTrace;

namespace QuickTrace.Examples;

public static class BasicExample
{
    public static async Task Run()
    {
        Console.WriteLine("📝 Basic Example");
        Console.WriteLine("================");
        Console.WriteLine();

        // Basic usage với default style
        var tracer = Tracer.NewSimpleTracer("Basic Example");

        await Task.Delay(30);
        tracer.Span("Initialize database");

        await Task.Delay(50);
        tracer.Span("Load user data");

        await Task.Delay(20);
        tracer.Span("Process data");

        await Task.Delay(10);
        tracer.Span("Generate response");

        tracer.End();
    }
}
